runtime/compiler/optimizer/UnsafeFastPath.cpp

```diff
@@ -142,16 +154,24 @@ void TR_UnsafeFastPath::transformUnsafeAtomicCallInVarHandleAccessMethod(TR::Tre
       {
       // codegen doesn't optimize CAS on a static field
       //
-      if (callerMethod != TR::java_lang_invoke_StaticFieldVarHandle_StaticFieldVarHandleOperations_OpMethod)
+      if (callerMethod != TR::java_lang_invoke_StaticFieldVarHandle_StaticFieldVarHandleOperations_OpMethod &&
+         performTransformation(comp(), "%s transforming Unsafe.CAS [" POINTER_PRINTF_FORMAT "] into codegen inlineable\n", optDetailString(), node))
          {
          node->setIsSafeForCGToFastPathUnsafeCall(true);
+         if (!isVarHandleOperationMethodOnArray(callerMethod))
+            {
+            node->setUnsafeGetPutCASCallOnNonArray();
+            }
+
          if (trace())
             {
             traceMsg(comp(), "Found Unsafe CAS node %p n%dn on non-static field, set the flag\n", node, node->getGlobalIndex());
             }
+
+         return true;
          }
 
-      return;
+      return false;
```

Comment:
The code following this point is dead. Was this intentional?

Reply:
Yes. We don't have intrinsics for Unsafe CAS right now. The flag is a workaround and CAS will be processed as the same way once the CAS atomic symbol is in. For that I'll create a separate PR.

Reply:
Ok, I'd like to see a TODO left here describing exactly why we have a dead end point here. The TODO should reference an issue (for example `TODO (#1234): Temporarily disable the following code until CAS work is done`) which will solve the problem.

Reply:
Sure, I've created https://github.com/eclipse/openj9/issues/3532 as the follow-up item.